### PR TITLE
Add vision evaluation metrics (exact_match, relaxed_accuracy, word_sort_ratio)

### PR DESCRIPTION
**Pull request by jiafatom**
## Summary

Extends Olive's evaluator framework with three vision-oriented accuracy sub-metrics for VQA, ChartQA, and OCR evaluation, following the existing pattern used for speech metrics (PR #2444).

## New Metrics

| Metric | Task Type | Suitable Benchmarks |
|--------|-----------|-------------------|
| `exact_match` | `vision-vqa` | AI2D, ScienceQA, TextVQA, MathVista, MMMU, InterGPS |
| `relaxed_accuracy` | `vision-chart-qa` | ChartQA (±5% numeric tolerance for numbers) |
| `word_sort_ratio` | `vision-ocr` | OCR benchmarks (word-level overlap) |

## Public HuggingFace Datasets

These metrics are designed to work with publicly available datasets:

| Metric | Recommended Dataset | HuggingFace ID |
|--------|-------------------|----------------|
| `exact_match` | TextVQA | `facebook/textvqa` |
| `relaxed_accuracy` | ChartQA | `HuggingFaceM4/ChartQA` |
| `word_sort_ratio` | DocumentVQA | `HuggingFaceM4/DocumentVQA` |

Example configuration snippets are provided in `docs/source/how-to/configure-workflows/metrics-configuration.md`.

## Changes

- **`olive/evaluator/metric.py`**: Adds `EXACT_MATCH`, `RELAXED_ACCURACY`, `WORD_SORT_RATIO` to `AccuracySubType` enum
- **`olive/evaluator/accuracy.py`**: Implements the three metric classes with multi-answer support
- **`olive/evaluator/olive_evaluator.py`**: Adds vision inference path and task-metric validation
- **`olive/data/component/pre_process_data.py`**: Adds `vision_vqa_pre_process` component
- **`olive/data/component/dataloader.py`**: Adds `vision_vqa_dataloader` with custom collate_fn for PIL images
- **`olive/data/container/huggingface_container.py`**: Registers `vision-vqa`, `vision-chart-qa`, `vision-ocr` task types with appropriate dataloader
- **`olive/olive_config.json`**: Adds `vision` extras (pillow)
- **`docs/source/how-to/configure-workflows/metrics-configuration.md`**: Adds vision metrics documentation with public dataset examples
- **`test/evaluator/test_accuracy.py`**: Unit tests covering all new metrics

## Design

- Vision metrics are text-based (compare predicted answer string to ground truth), task-dependent
- Multiple valid answers supported via `|` separator (metrics match against any valid answer)
- Task-metric validation ensures incompatible combinations raise `ValueError`
- Custom `vision_vqa_dataloader` handles PIL images with a collate_fn that avoids PyTorch default collation issues
- PyTorch path: model processor handles images natively
- ONNX path: single forward pass assumed (classification-style VQA); for autoregressive models use PyTorch evaluator with generation loop

### CODE REVIEW

File: olive/data/component/pre_process_data.py [hunk 1]
```diff
@@ -379,3 +379,92 @@ def collate_fn(batch):
...
+        def __len__(self):
+            return len(self.dataset)
+
+        def __getitem__(self, idx):
+            item = self.dataset[idx]
+            image = item[self.image_column]
+            question = item[self.question_column]
+            answer = item[self.answer_column]
+            # Handle list answers (some datasets have multiple valid answers)
+            if isinstance(answer, list):
```
Comment (shaahji) [inline olive/data/component/pre_process_data.py]: list or tuple

File: olive/data/component/pre_process_data.py [hunk 2]
```diff
@@ -379,3 +379,92 @@ def collate_fn(batch):
...
+                answer = answer[0] if answer else ""
+            return {"image": image, "question": question}, str(answer)
+
+        @staticmethod
+        def collate_fn(batch):
+            """Collate VQA batches. Use with batch_size=1 for variable-size images."""
+            if len(batch) == 1:
+                input_dict, answer = batch[0]
+                return (input_dict, [answer])
```
Comment (shaahji) [inline olive/data/component/pre_process_data.py:470]: How are "multiple valid answers" handled here? Wouldn't this create a _list of lists_ thus making it a bad input downstream?

File: olive/evaluator/accuracy.py [hunk 3]
```diff
@@ -217,3 +217,175 @@ def measure(self, model_output, target):
...
+            pred_is_num, pred_val = self._try_parse_number(pred_str)
+            ref_is_num, ref_val = self._try_parse_number(ref_str)
+
+            if pred_is_num and ref_is_num:
+                # Numeric comparison with tolerance
+                if ref_val == 0:
+                    if pred_val == 0:
+                        correct += 1
+                elif abs(pred_val - ref_val) / abs(ref_val) <= tolerance:
+                    correct += 1
```
Comment (shaahji) [inline olive/evaluator/accuracy.py]: nit: Simplify this snippet

```
if (ref_val == pred_val) or (abs(pred_val - ref_val) / abs(ref_val) <= tolerance):
    correct += 1
```

File: olive/evaluator/olive_evaluator.py [hunk 4]
```diff
@@ -646,6 +718,74 @@ def _inference_text(
...
+
+        all_preds = []
+        all_targets = []
+        output_names = io_config["output_names"]
+        is_single_tensor_output = len(output_names) == 1
+
+        for batch in dataloader:
+            input_data, labels = OliveEvaluator.unpack_batch_for_accuracy(batch)
+            input_feed = format_data(input_data, io_config)
+            result = model.run_session(session, input_feed, **run_kwargs)
```
Comment (jambayk) [inline olive/evaluator/olive_evaluator.py:758]: curious if there are are models that can generate the answer just with a single session run. are these model's conditional generation models + don't need a generation loop?

**Review by shaahji — CHANGES_REQUESTED** [review]

**Comment by shaahji** [comment]
@jiafatom Closing this PR. Please open a new one directly in Olive repo. PRs from forked repos can't be built and verified without additional support from core others. PRs in Olive repo auto run the verification process.

**Comment by jiafatom** [comment]
> @jiafatom Closing this PR. Please open a new one directly in Olive repo. PRs from forked repos can't be built and verified without additional support from core others. PRs in Olive repo auto run the verification process.

I thought this is from Olive repo? https://github.com/microsoft/Olive/tree/jiafa/add-vision-eval-metrics

**Review by shaahji — APPROVED** [review]
